docs/PremiumApps.md

```diff
@@ -0,0 +1,488 @@
+4. Place the extracted Splunk_TA_ForIndexers package in the s3 bucket folder named "es_app_indexer_ta"
+
+The operator will poll this bucket after configured appsRepoPollIntervalSeconds and install the Splunk_TA_ForIndexers also.
+ 
+In this example for SHC, scope=premiumApps, type=EnterpriseSecurity, sslEnablement=ignore
+and for ClusterManager, scope=cluster, type and sslEnablement are not applicable.
+
+```yaml
+apiVersion: enterprise.splunk.com/v4
+kind: SearchHeadCluster
+metadata:
+  name: shc-es
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  appRepo:
+    appsRepoPollIntervalSeconds: 60
+    defaults:
+      volumeName: volume_app_repo
+      scope: local
+    appSources:
+      - name: esApp
+        location: es_app/
+        scope: premiumApps
+        premiumAppsProps:
+          type: enterpriseSecurity
+          esDefaults:
+             sslEnablement: ignore
+    volumes:
+      - name: volume_app_repo
+        storageType: s3
+        provider: aws
+        path: security-team-apps/
+        endpoint: https://s3-us-west-2.amazonaws.com
+        region: us-west-2
+        secretRef: splunk-s3-secret
+  ClusterManagerRef:
+    name: cm-es
+---
+apiVersion: enterprise.splunk.com/v4
+kind: ClusterManager
+metadata:
+  name: cm-es
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  appRepo:
+    appsRepoPollIntervalSeconds: 60
+    defaults:
+      volumeName: volume_app_repo
+      scope: local
+    appSources:
+      - name: esAppIndexer
+        location: es_app_indexer_ta/
+        scope: cluster
+    volumes:
+      - name: volume_app_repo
+        storageType: s3
+        provider: aws
+        path: security-team-apps/
+        endpoint: https://s3-us-west-2.amazonaws.com
+        region: us-west-2
+        secretRef: splunk-s3-secret
+---
+apiVersion: enterprise.splunk.com/v2
+kind: IndexerCluster
+metadata:
+  name: idc-es
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  ClusterManagerRef:
+    name: cm-es
+  replicas: 3
+```
+
+#### Special consideration while using ssl enabled mode of strict in SHC
+
+For using the strict mode, the following additional steps are required so that SHC has Splunk Web SSL enabled. These steps are necessary before you can install ES app with the strict mode of sslEnablement. Alternatively, if you are managing enableSplunkWebSSL setting outside of the Operator scope by pushing an app bundle through the deployer, then you can skip all these steps.  
+
+Steps:
+1. Create a shc app (e.g., shccoreapp.spl) that contains a local/web.conf setting with `enableSplunkWebSSL=true`
+2. Place this app under security-team-apps/coreapps in your s3 bucket.
+3. Use the following YAML file as an example to deploy this app through app framework. 
+4. Note: you also need to include the extraEnv:SPLUNK_HTTP_ENABLESSL with the value true in the YAML. This step will be revised in upcoming releases so it may not be needed as you are already pushing a web.conf with ssl setting.
```

Comment:
Step 4 would be deleted if the [ansible fix](https://github.com/splunk/splunk-ansible/pull/704) -" to not set enableSplunkWebSSL to false in system/local/web.conf in SHs by default" is released before Operator 2.2 release.
Also remove the extraEnv:SPLUNK_HTTP_ENABLESSL entries from SHC YAML examples (line 379,380, 381)

Reply:
We will fix this once the ansible PR is merged.